api/src/main/java/org/openmrs/module/fhir2/api/impl/FhirTaskServiceImpl.java

```diff
@@ -71,5 +78,17 @@ public Task updateTask(String uuid, Task task) {
 		
 		return translator.toFhirResource(dao.saveTask(translator.toOpenmrsType(openmrsTask, task)));
 	}
-	
+
+	@Override
+	public Collection<Task> getTasksByBasedOn(Class<? extends DomainResource> obj, String uuid) {
+		Collection<Task> associatedTasks = new ArrayList<>();
+
+		if(obj == ServiceRequest.class || obj == MedicationRequest.class || obj == CarePlan.class){
+			for(org.openmrs.module.fhir2.Task openmrsTask : dao.getTasksByBasedOnUUID(obj.getSimpleName(), uuid)){
+				associatedTasks.add(translator.toFhirResource(openmrsTask));
```

Comment:
This could also be:

`dao.getTaskByBasedOnUuid(obj.getSimpleName(), uuid).stream().map(translator::toFhirResource).collect(Collectors.toList())`